DPG/Tasks/qaMatchEff.cxx

```diff
@@ -420,8 +478,32 @@ struct qaMatchEff {
           //
       }   // end pions only
       //
+      // no primary/sec-d pions
+      if (!((tpPDGCode == 211) && (mcpart.isPhysicalPrimary()))) {
```

Comment:
here you are requiring that the particle is a Physical Primary, but not a pion. Is it the combination you want?